internal/backend/remote-state/s3/backend.go

```diff
@@ -509,6 +612,40 @@ func getS3Config(obj cty.Value) func(options *s3.Options) {
 	}
 }
 
+func configureNestedAssumeRole(obj cty.Value) *awsbase.AssumeRole {
+	assumeRole := awsbase.AssumeRole{}
+
+	obj = obj.GetAttr("assume_role")
+	if val, ok := stringAttrOk(obj, "role_arn"); ok {
+		assumeRole.RoleARN = val
+	}
+	if val, ok := stringAttrOk(obj, "duration"); ok {
+		dur, _ := time.ParseDuration(val)
```

Comment:
Shouldn't this error be handled?

Reply:
Oh, now I see it's handled in the validation function.

In that case, I'd panic here on error.

Reply:
We know that this is always possible to parse because we check it beforehand: https://github.com/opentofu/opentofu/pull/754/files#diff-c6f102657948029362a41861b9418f65f3830155d90e53b4a5141ddb2d60c2b4R80

Reply:
unless it is misused, which is why I propose the panic, just in case